Adjust hidden usage

leomoty opened the pull request:
For #13207. 

Since this is not new code is keeping jQuery fine for now?

Also I still have to check the other modules, I mostly wanted to know if this was the direction @garrett wanted to go.

leomoty commented:
This should also fix #13200

garrett commented:
My big concern with the branch is if `prop("hidden", false);` actually removes the value, as hidden is [a boolean attribute](https://www.w3.org/TR/html51/infrastructure.html#boolean-attribute), it doesn't accept true/false.

However, I'm not sure what `.prop()` does in this case. (I haven't checked yet, as I'm trying to wrap up some other PRs before going on vacation at the moment.)

Regardless, if it does need to be adjusted, I still think this is a good base to build upon.

leomoty commented:
We could use attr/removeAttr, or prop/removeProp I guess. I don't think the browsers keep the true/false portion nonetheless.

leomoty commented:
I've also tried to skip any show/hide that did not have a hidden attribute before, with exception of that danger inline-alert from NetworkManager.

garrett commented:
@leomoty: Yeah, [the jQuery docs](https://api.jquery.com/prop/) make it look like it might do the right thing. It's just something we should double-check when the code is running.

leomoty commented:
@garrett well I did test it locally, the attribute is attached and removed from DOM representation properly, at least on recent Firefox/Chrome.

leomoty commented on pkg/networkmanager/interfaces.js:
```diff
@@ -4650,7 +4650,7 @@ function dialog_setup(d) {
     $('#' + d.id)
             .on('show.bs.modal', function () { d.enter() })
-            .on('shown.bs.modal', function () { d.show() })
+            .on('shown.bs.modal', function () { d.prop('hidden', false) })
```
oops, this is definitely wrong.

martinpitt commented:
Thanks @leomoty , this generally looks nice! Apparently you found something wrong in networkmanager still. Once you force-push, can you please:

 - Add "Fixes #13273" to either the first ("most important") or top-most commit
 - Add "Closes #13265" to the topmost commit
 - Prefix the second commit with "shell:"

> Since this is not new code is keeping jQuery fine for now?

Yes, of course.

> How do I access the playground?

That's the "Development" menu entry. You usually only see it when running from the built tree (which you do, presumably).

> [sos-intro] This is not used anywhere? 

Indeed, this can go.

This is still marked as "draft". What's still missing from your POV?

I start a couple of tests now, as certainly there will be some fallout.

leomoty commented:
@martinpitt At least systemd is missing, I think. Should also take a look again at realmd but I think the hide/show usage there is unrelated to the hidden elements.

leomoty commented:
> * Add "Fixes #13273" to either the first ("most important") or top-most commit
> * Add "Closes #13265" to the topmost commit
> * Prefix the second commit with "shell:"

Done

> > [sos-intro] This is not used anywhere?

Got rid of it

> I start a couple of tests now, as certainly there will be some fallout.

Seems like I lost the tests results after force push? I'll remove the draft, but I am pretty sure there are some missing cases. Like, I just found "Active Pages".

leomoty commented:
I don't exactly know how to run the tests, and would love for a primer, but i'll keep a list of what I think I fixed and will push as FIXUPs for now:


- [x] testBasic (check_connection.TestConnection)

marusak commented:
Ok, so firstly [here](https://209.132.184.41:8493/logs/pull-13265-20191211-194848-0987d821-cockpit-project-cockpit--fedora-30/log.html) is fedora-30 log from previous run to have something to compare.

How to run tests: (it is very well documented in [README](https://github.com/cockpit-project/cockpit/blob/master/test/README.md))
1. `./test/image-prepare fedora-30`
2. `./test/verify/check-*`

You want to run only specific tests, so what I do is check some failed test, like for example [this one](https://209.132.184.41:8493/logs/pull-13265-20191211-194848-0987d821-cockpit-project-cockpit--fedora-30/log.html#40-2). It name is `testResources (check_docker.TestDocker)`, so I know I need to run `./test/verify/check_docker TestDocker.testResources`. If you don't want to construct the name like this, just do `./test/verify/check_docker -l` and then copy the appropriate name.

Also you most likely want to run it with `-st` to see some more verbose logging. 
Also you might want interactive tests, so check in the readme how we use `chromium-browser`.
Also super useful thing is to use `sit()` :) Also mentioned in Readme.

martinpitt commented:
@leomoty : Next time you push, would you mind rebasing to current master? That will fix the semaphore error. Thanks! (Please feel free to squash your fixups on the occasion, of course)

leomoty commented:
> @leomoty : Next time you push, would you mind rebasing to current master? That will fix the semaphore error. Thanks! (Please feel free to squash your fixups on the occasion, of course)

Rebased and squashed to their specific counterparts, except the sos-intro.

leomoty commented:
[previous logs](https://209.132.184.41:8493/logs/pull-13265-20191213-111414-c0fd020a-cockpit-project-cockpit--fedora-30/log.html)

@marusak,

* Menu one seems unrelated
* realms, I have fixed and works locally
* system info, also fixed and works.

Can you take a look at networking ones? That is all that is left.

leomoty commented:
@marusak mind spinning f30 again? I cannot get those tests to finish locally. But for other reasons (mirrors.fedoraproject.org could not be resolved, I guess?).

leomoty commented:
@martinpitt @marusak I think this is ready for a review.

martinpitt commented:
@leomoty : This looks great now, thanks! I'm happy to see that this doesn't even need test changes, i. e. `wait_visible()` and such is a good enough abstraction for this. Can you please squash the fixups? Or want us to do that? (Happy to help with that, just asking)

leomoty commented:
@martinpitt, if you have the time do it, won't be able to touch it for at least 5-6 hours. Also, I guess we should add a Fixes to the issue #13200 on the respective commit (552d5ebc39ef78299eec4ea2eaeb43a8e464444e).

leomoty commented:
Also, being honest, I am impressed how thorough the E2E tests are, after understanding how `prepare-image` works it was a breeze to work with.

martinpitt commented:
@leomoty : Rebased and squashed, and Fixes # added. Thanks a lot for this, great work!
